wire-ios-sync-engine/Source/Use cases/EnableAnalyticsUseCase.swift

```diff
@@ -23,79 +23,28 @@ import WireAnalytics
 
-    /// Invokes the use case to enable analytics sharing.
-    func invoke()
+    /// Enable analytics tracking.
+
+    func invoke() throws
+
 }
 
 // MARK: - EnableAnalyticsUseCase
 
-/// Concrete implementation of the EnableAnalyticsUseCaseProtocol.
-/// This struct is responsible for enabling analytics sharing for a specific user profile.
 struct EnableAnalyticsUseCase: EnableAnalyticsUseCaseProtocol {
 
-    // MARK: - Properties
-
-    /// The configuration for the analytics session.
-    private(set) var analyticsSessionConfiguration: AnalyticsSessionConfiguration
-
-    /// A closure that provides an `AnalyticsManagerProtocol` implementation.
-    private(set) var analyticsManagerBuilder: (_ appKey: String, _ host: URL) -> any AnalyticsManagerProtocol
-
-    /// The session manager that conforms to `AnalyticsManagerProviding` for managing analytics sessions.
-    private(set) var sessionManager: AnalyticsManagerProviding
-
-    /// The user profile for which to enable analytics sharing.
-    private(set) var analyticsUserProfile: AnalyticsUserProfile
-
-    /// An instance conforming to `EnableAnalyticsUseCaseAnalyticsSessionProviding` that manages the user's analytics session state.
-    private(set) var analyticsSessionProvider: EnableAnalyticsUseCaseAnalyticsSessionProviding
+    let currentUser: AnalyticsUser?
+    let service: any AnalyticsServiceProtocol
 
-    // MARK: - Initialization
+    func invoke() throws {
+        try service.enableTracking()
 
-    /// Initializes a new instance of EnableAnalyticsUseCase.
-    ///
-    /// - Parameters:
-    ///   - analyticsManagerBuilder: A closure that provides an `AnalyticsManagerProtocol` implementation.
-    ///   - sessionManager: The session manager that conforms to `AnalyticsManagerProviding` for managing analytics sessions.
-    ///   - analyticsSessionConfiguration: The configuration for the analytics session.
-    ///   - analyticsUserProfile: The user profile for which to enable analytics sharing.
-    ///   - analyticsSessionProvider: An instance conforming to `EnableAnalyticsUseCaseAnalyticsSessionProviding` that manages the user's analytics session state.
-    init(
-        analyticsManagerBuilder: @escaping (_ appKey: String, _ host: URL) -> any AnalyticsManagerProtocol,
-        sessionManager: AnalyticsManagerProviding,
-        analyticsSessionConfiguration: AnalyticsSessionConfiguration,
-        analyticsUserProfile: AnalyticsUserProfile,
-        analyticsSessionProvider: EnableAnalyticsUseCaseAnalyticsSessionProviding
-    ) {
-        self.analyticsManagerBuilder = analyticsManagerBuilder
-        self.sessionManager = sessionManager
-        self.analyticsSessionConfiguration = analyticsSessionConfiguration
-        self.analyticsUserProfile = analyticsUserProfile
-        self.analyticsSessionProvider = analyticsSessionProvider
+        if let currentUser {
+            service.switchUser(currentUser)
+        }
```

Comment:
We may enable analytics before there exists a current user, therefore it's optional.